src/hooks/users.ts

```diff
@@ -85,24 +90,59 @@ export const useWorldUsersByIdWorkaround = () => {
   return { worldUsersById, isWorldUsersLoaded };
 };
 
+export const useWorldUserLocation = (
+  userId?: string
+): { userLocation?: WithId<UserLocation> } => {
+  useConnectWorldUsers();
+
+  const userLocation = useSelector((state) => {
+    if (!userId) return;
+
+    const user = worldUsersByIdSelector(state)?.[userId];
+
+    if (!user) return;
+
+    const userLocation: UserLocation = {
+      lastSeenAt: user.lastSeenAt,
+      lastSeenIn: user.lastSeenIn,
+    };
+
+    return withId(userLocation, userId);
```

Comment:
Maybe we should rename this function to start with a verb instead, because it's slightly miss-leading atm. It just adds/merges the second provided parameter as an `id` field into the object provided as first parameter.

Reply:
I don't understand your issue with it? Also, this is code that's existed and used everywhere across the platform for months. It's a helper that corresponds with the `WithId` type, that turns `Foo` into `WithId<Foo>`